interface-impl-amf/src/main/java/org/mule/amf/impl/model/ParameterValidationStrategyBuilder.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.amf.impl.model;
+
+import amf.client.model.domain.AnyShape;
+import amf.client.model.domain.ArrayShape;
+import amf.client.model.domain.ScalarShape;
+import amf.client.model.domain.UnionShape;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Set;
+
+class ParameterValidationStrategyBuilder {
```

Comment:
I would change it by a factory instead, where the Shape is the discriminator passed as parameter in a `getStrategy()` method.

Reply:
agree, will do.